homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,129 @@
+    BooleanSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from . import SlideConfigEntry
+from .const import CONF_INVERT_POSITION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BOOLEAN_SELECTOR = BooleanSelector()
+
+API_VERSION_SELECTOR = SelectSelector(
+    SelectSelectorConfig(
+        options=[
+            SelectOptionDict(value="1", label="API 1"),
+            SelectOptionDict(value="2", label="API 2"),
+        ],
+        mode=SelectSelectorMode.DROPDOWN,
+        translation_key=CONF_API_VERSION,
+    )
+)
+
+
+class SlideConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for slide_local."""
+
+    _entry: SlideConfigEntry | None = None
+    _mac: str = ""
+    _host: str = ""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    async def async_test_connection(
+        self, user_input: dict[str, str | int]
+    ) -> dict[str, str]:
+        """Reusable Auth Helper."""
+        slide = SlideLocalApi()
+        await slide.slide_add(
+            user_input[CONF_HOST],
+            user_input[CONF_PASSWORD],
+            user_input[CONF_API_VERSION],
+        )
+
+        try:
+            result = await slide.slide_info(user_input[CONF_HOST])
+            self._mac = result.get("mac", "")
+        except (ClientConnectionError, ClientTimeoutError):
+            return {"base": "cannot_connect"}
+        except (AuthenticationFailed, DigestAuthCalcError):
+            return {"base": "invalid_auth"}
+        except Exception as e:  # noqa: BLE001
+            _LOGGER.error(e)
+            return {"base": "unknown"}
+
+        return {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the user step."""
+        errors = {}
+        if user_input is not None and user_input.get(CONF_API_VERSION) is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            user_input[CONF_API_VERSION] = int(user_input[CONF_API_VERSION])
+
+            if not (errors := await self.async_test_connection(user_input)):
+                user_input |= {CONF_MAC: format_mac(self._mac)}
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        if user_input is not None and user_input.get(CONF_HOST) is not None:
+            self._host = user_input[CONF_HOST]
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._host): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Required(CONF_API_VERSION, default="2"): API_VERSION_SELECTOR,
+                    vol.Required(CONF_INVERT_POSITION, default=False): BOOLEAN_SELECTOR,
+                }
+            ),
+            errors=errors,
```

Comment:
For the zeroconf, Ideally we don't even ask them the host anymore as this way they can change it again